runtime/jcl/common/criusupport.c

```diff
@@ -283,16 +318,25 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 		resultType = vm->criuSupportSuccess;
 
 releaseExclusive:
+		vmFuncs->releaseExclusiveVMAccess(currentThread);
+closeWorkDirFD:
+		if (0 != close(workDirFD)) {
+			systemReturnCode = errno;
+			resultType = vm->criuSupportJVMCheckpointFailure;
+			vmFuncs->setCurrentExceptionNLSWithArgs(currentThread, J9NLS_JCL_CRIU_FAILED_TO_CLOSE_DIR, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, systemReturnCode);
+		}
+freeWorkDir:
 		if (0 != close(dirFD)) {
 			systemReturnCode = errno;
-			resultType = vm->criuSupportJVMRestoreFailure;
+			resultType = vm->criuSupportJVMCheckpointFailure;
```

Comment:
this may fail after checkpoint as well so I think youll need to create a variable that lets you know if this is coming from pre or post checkpoint

Reply:
But wouldn't the source of the problem always be before checkpoint? the directory is not used post checkpoint/restore so it wouldn't be a "restore" failure

Reply:
Not necessarily, there is a window between when the checkpoint happens and when the restore occurs. In that window a number of things may change in the underlying filesystem that may cause `close(..` to fail.

Reply:
So do I set a variable here e.g. afterCheckpoint to differentiate? https://github.com/eclipse-openj9/openj9/pull/13185/files#diff-2d72f5530c1f3f0e677aecd17b0123fba8cef14d8dda553e6a7dd2993b4171d4R312

Reply:
This also applies to the close(workDirFD) as well right

Reply:
> So do I set a variable here e.g. afterCheckpoint to differentiate? line 312
> This also applies to the close(workDirFD) as well right

yes

Reply:
Ok done